src/listeners/message-listeners/message-handler.ts

```diff
@@ -17,8 +17,12 @@ export interface ActionHandler<Data> {
  */
 export function createMessageHandler(
   handlers: ActionHandler<unknown>[],
-): (action: BackgroundAction, data: unknown) => Promise<unknown> {
-  return (action: BackgroundAction, data: unknown): Promise<unknown> => {
+): (action: BackgroundAction, data: unknown, sender: chrome.runtime.MessageSender) => Promise<unknown> {
+  return (
+    action: BackgroundAction,
+    data: unknown,
+    sender: chrome.runtime.MessageSender,
+  ): Promise<unknown> => {
     const handlerObject = handlers.find((handler) => handler.action === action)
 
     if (!handlerObject) {
```

Comment:
This is just an instance of message handler. It checks if the received message is intended for it. If it isn't, it will just ignore it.
In the `src/listeners/message-listeners/index.ts` file, the function checks if any of the existing message handlers responded, and if not, then an error is thrown.